src/threeIdProvider.ts

```diff
@@ -19,94 +20,98 @@ type Context = {
 }
 
 const methods: HandlerMethods<Context> = {
-  '3id_getLink': async (ctx) => {
-    return (await ctx.provider.wallet.getLink()).toLowerCase()
-  },
-  '3id_linkManagementKey': async (ctx) => {
-    return await ctx.provider.wallet.linkManagementKey()
-  },
-  '3id_authenticate': async (ctx, params) => {
-    return await ctx.provider.wallet.authenticate(
-      params.spaces,
-      {
-        authData: params.authData,
-        address: params.address,
-        mgmtPub: params.mgmtPub,
-      },
-      ctx.origin
-    )
+  '3id_authenticate': async ({ permissions, keyring, origin }, params) => {
+    const spaces = await permissions.request(origin, params.spaces || [])
+    return {
+      main: keyring.getPublicKeys({ mgmtPub: params.mgmtPub }),
+      spaces: spaces.reduce((acc, space) => {
+        acc[space] = keyring.getPublicKeys({
+          space,
+          uncompressed: true,
+        })
+        return acc
+      }, {} as Record<string, PublicKeys>),
+    }
   },
-  '3id_isAuthenticated': async (ctx, params) => {
-    return await ctx.provider.wallet.isAuthenticated(params.spaces, ctx.origin)
+  '3id_isAuthenticated': async ({ permissions, origin }, params) => {
+    return permissions.has(origin, params.spaces)
   },
-  '3id_signClaim': async (ctx, params) => {
-    return await ctx.provider.wallet.signClaim(params.payload, {
-      DID: params.did,
-      space: params.space,
+  '3id_signClaim': async ({ threeIdx, permissions, keyring, origin }, params) => {
+    if (!permissions.has(origin, params.spaces)) {
+      throw new RPCError(0, 'Authentication required')
```

Comment:
We should probably replace these by the error codes and messages defined in https://eips.ethereum.org/EIPS/eip-1193 so better consistency